### DISCUSSION TITLE
After binding ListBox.SelectedItemProperty, as if the update of the bound value was after the SelectionChanged event was triggered?

### DISCUSSION THREAD

**Discussion by WenLiHou** [AvaloniaUI/Avalonia#13686] [General]
After binding ListBox.SelectedItemProperty, in the SelectionChanged event, the value of ListBox.SelectedItem is inconsistent with the bound property, as if the update of the bound value was after the SelectionChanged event was triggered, which caused me to not be able to bind passing CommandParameters when creating a custom Command in custom Control.

**Comment by WenLiHou** [comment]
`
  public HomeView()
  {
      InitializeComponent();


      listbox.ItemsSource = new List<string> { "a", "b", "c" };

      listbox.Bind(ListBox.SelectedItemProperty, new Binding
      {
          Source = this,
          Path = nameof(TestItem)
      });

      listbox.SelectionChanged += Listbox_SelectionChanged;
  }

  public object? TestItem { get; set; }


  private void Listbox_SelectionChanged(object? sender, SelectionChangedEventArgs e)
  {

      var a = listbox.SelectedItem;

      var b = TestItem;
  }
`

**Reply by WenLiHou** [reply]
The values of a and b are not equal.

**Reply by timunie** [reply]
TestItem needs to raise PropertyChanged, otherwise the binding does not work correctly.

**Reply by WenLiHou** [reply]
The value of the TestItem changes, but each time the value of the TestItem is read in the SelectionChanged event is the  past selected value, causing it to be different from the value of the listbox.SelectedItem

**Reply by timunie** [reply]
TestItem need to report its new value back. So it needs to send a PropertyChanged-event but it does not acc. to your snipped

**Reply by WenLiHou** [reply]
![123456](https://github.com/AvaloniaUI/Avalonia/assets/39956314/1d4b5eb8-6e41-4e82-9b91-05a6f266e905)

last time i selected “a”, this time i selected "c" ,the value of TestItem updated after SelectionChanged Event by Avalonia

**Reply by WenLiHou** [reply]
If the next time I select “b”, the TestItem is updated to “c”

**Reply by MrJul** [reply]
Can you please try latest nightly builds? This should have been fixed by #13503.

**Reply by WenLiHou** [reply]
I tried，it's ok.
